File: 1512.number-of-good-pairs.java
/*
 * @lc app=leetcode id=1512 lang=java
 *
 * [1512] Number of Good Pairs
 */

// @lc code=start

import java.util.HashMap;
import java.util.Map;

class Solution {
    public int numIdenticalPairs(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int n : nums) {
            map.put(n, map.getOrDefault(n, 0) + 1);
        }
        int ret = 0;
        for (Map.Entry<Integer, Integer> e : map.entrySet()) {
            int times = e.getValue();
            if (times > 1) {
                ret += (times * (times - 1) / 2);
            }
        }
        return ret;
    }
}
// @lc code=end

File: 1512.number-of-good-pairs.cpp
/*
 * @lc app=leetcode id=1512 lang=cpp
 *
 * [1512] Number of Good Pairs
 */

// @lc code=start
#include <vector>
#include <array>

using namespace std;

class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        array<unsigned, 100> umap {};
        for (const auto &el : nums) ++umap[el - 1];
        unsigned cnt = 0;
        for (const auto &el : umap) {
            if (el > 1) cnt += el * (el - 1) / 2;
        }
        return cnt;
    }
};
// @lc code=end
